lib/galaxy/security/validate_user_input.py

```diff
@@ -45,7 +45,7 @@ def validate_publicname( trans, publicname, user=None ):
         if len( publicname ) < 3:
             return "Public name must be at least 3 characters in length"
     else:
-        if len( publicname ) < 4:
+        if len( publicname ) < 3:
             return "Public name must be at least 4 characters in length"
```

Comment:
see comments in PR, our LDAP allows 3 character usernames (as do many others), and if my memory is correct (we've since changed a few things) REMOTE_USER forced `esr`into the database despite Galaxy not explicity permitting it. I don't remember my public-name ending with a `-`.